crates/wysiwyg/src/dom/dom_list_methods.rs

```diff
@@ -152,6 +152,31 @@ where
         }
         self.join_nodes_in_container(&handle.parent_handle());
     }
+
+    /// Slice list item at given handle and offset. A ZWSP node
+    /// is added at the beginning of the created list item.
+    ///
+    /// * `handle` - the list item handle.
+    /// * `offset` - offset at which the list item should be sliced
+    pub(crate) fn slice_list_item(
+        &mut self,
+        handle: &DomHandle,
+        offset: usize,
+    ) {
+        let list_item = self.lookup_node_mut(handle);
+        let mut slice = list_item.slice_after(offset);
+        slice.as_container_mut().unwrap().add_leading_zwsp();
+        let list = self.lookup_node_mut(&handle.parent_handle());
+        let DomNode::Container(list) = list else { panic!("List node is not a container") };
+        if slice.text_len() == 0 {
+            list.insert_child(
+                handle.index_in_parent() + 1,
+                DomNode::new_list_item(vec![DomNode::new_zwsp()]),
+            );
+        } else {
+            list.insert_child(handle.index_in_parent() + 1, slice);
+        }
```

Comment:
Are the `+1`s here because of the presence of ZWSP? 

Also as we're calling the same method on list (but with different second arguments) do you think it might be easier to read if we do something like (may be syntax errors here)
```
let child = if slice.text_len() != 0 { slice } else { DomNode::new_list_item(vec![DomNode::new_zwsp()]) }

...

list.insert_child(..., child);
``` 

Maybe you think that's less clear, could just be a personal preference

Reply:
Thanks, I actually just removed this condition by making the `container_node` method for adding ZWSP work a bit better. 
The +1 is because we want to insert the new list item after the current one.